tests/ui/redundant_pattern_matching_option.rs

```diff
@@ -95,3 +97,8 @@ const fn issue6067() {
         None => true,
     };
 }
+
+#[allow(clippy::deref_addrof)]
+fn issue7921() {
+    if let None = *(&None::<()>) {}
```

Comment:
Does this also work for 

```suggestion
    if let None = *&None::<()> {}
```

I suspect, that the suggestion for this will be `if &None::<()>.is_none() {`, which will produce a compile error.

If that's the case, we should use our `clippy_utils::Sugg` struct with `.maybe_paren()`.